compiler/rustc_codegen_ssa/src/mir/constant.rs

```diff
@@ -84,11 +94,11 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                             };
                             bx.scalar_to_backend(prim, scalar, bx.immediate_backend_type(layout))
                         } else {
-                            bug!("simd shuffle field {:?}", field)
+                            bug!("field is not a scalar {:?}", field)
                         }
                     })
                     .collect();
-                bx.const_struct(&values, false)
+                if ty_is_simd { bx.const_vector(&values) } else { bx.const_struct(&values, false) }
```

Comment:
I think making this always a `const_vector` should work. The simd_shuffle code during codegen uses `LLVMGetAggregateElement` to access this which works both for structs and vectors. So you should be able to remove this part of the `ty_is_simd` hack.

(The part further up this function is still required, that will be a bit more work to get rid of -- I plan to do that in future PRs.)

Reply:
Removing this has now made the pipeline fail, are we okay to put it back?